contracts/enclaves/AttestationAutherUpgradeable.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "./interfaces/IAttestationVerifier.sol";
+
+contract AttestationAutherUpgradeable is
+    Initializable  // initializer
+{
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IAttestationVerifier public immutable ATTESTATION_VERIFIER;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable ATTESTATION_MAX_AGE;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(IAttestationVerifier attestationVerifier, uint256 maxAge) {
+        ATTESTATION_VERIFIER = attestationVerifier;
+        ATTESTATION_MAX_AGE = maxAge;
+    }
+
+    struct EnclaveImage {
+        bytes PCR0;
+        bytes PCR1;
+        bytes PCR2;
+    }
+
+    mapping(bytes32 => EnclaveImage) public whitelistedImages;
```

Comment:
Should these be private and add a getter instead so that we enforce these are the only way to whitelist images or verify keys?

Reply:
How does OZ do it?

Reply:
Though I remember we had to copy the snapshot stuff just because things were private :laughing: